### PR TITLE
[UI] Manage view for AAVE multiply

### PR DESCRIPTION
**Pull request by marcinciarka**
# [[UI] Manage view for AAVE multiply](https://app.shortcut.com/oazo-apps/story/6438/ui-manage-view-for-aave-multiply)

  
## Changes 👷‍♀️
- added _Manage view for AAVE multiply_ (which includes non-connected view, connected view and managing view + opening view, everything is managed by the same component)
-  
## How to test 🧪
To test the component, view it on:
- _your_ aave multiply position
- _someones_ multiply position (I can provide a link if you want)
- opening aave steth multiply view

### CODE REVIEW

File: features/earn/aave/components/AavePositionHeader.tsx [hunk 1]
```diff
@@ -120,9 +120,11 @@ function AavePositionHeader({
 
 export function headerWithDetails(minimumRiskRatio: IRiskRatio) {
   return function AavePositionHeaderWithDetails({ strategyName }: { strategyName: string }) {
-    const { aaveTotalValueLocked$, aaveReserveStEthData$ } = useAaveContext()
+    const { aaveTotalValueLocked$, aaveReserveConfiguration } = useAaveContext()
     const [tvlState, tvlStateError] = useObservable(aaveTotalValueLocked$)
-    const [aaveReserveConfigData, aaveReserveConfigDataError] = useObservable(aaveReserveStEthData$)
+    const [aaveReserveConfigData, aaveReserveConfigDataError] = useObservable(
+      aaveReserveConfiguration['STETH'],
```
Comment (jakubswierczek) [inline features/earn/aave/components/AavePositionHeader.tsx:126]: token should be read from `strategyConfig`

File: features/multiply/aave/components/AaveMultiplyManageComponent.tsx [hunk 2]
```diff
@@ -1,5 +1,50 @@
...
-  return <>aave multiply manage section</>
+import { AaveMultiplyPositionData } from './AaveMultiplyPositionData'
+
+export function AaveMultiplyManageComponent({ isManage = false }: { isManage?: boolean }) {
+  const { stateMachine } = isManage
+    ? useManageAaveStateMachineContext()
+    : useOpenAaveStateMachineContext()
+  const [state] = useActor(stateMachine)
+  const { aaveReserveData } = useAaveContext()
+  const [aaveUSDCReserveData] = useObservable(aaveReserveData['USDC'])
```
Comment (jakubswierczek) [inline features/multiply/aave/components/AaveMultiplyManageComponent.tsx:17]: from config + add error handling

File: features/multiply/aave/components/AaveMultiplyManageComponent.tsx [hunk 3]
```diff
@@ -1,5 +1,50 @@
...
+import { AppSpinner, WithLoadingIndicator } from 'helpers/AppSpinner'
+import { useObservable } from 'helpers/observableHook'
 import React from 'react'
 
-export function AaveMultiplyManageComponent() {
-  return <>aave multiply manage section</>
+import { AaveMultiplyPositionData } from './AaveMultiplyPositionData'
+
+export function AaveMultiplyManageComponent({ isManage = false }: { isManage?: boolean }) {
+  const { stateMachine } = isManage
```
Comment (jakubswierczek) [inline features/multiply/aave/components/AaveMultiplyManageComponent.tsx:12]: common 😃
Try to pass it in props, or pass only meaningful things from the state machine's context. 
Right now the component is defined in `strategy config` and it's not aware of `useManageAaveStateMachineContext()`  or `useOpenAaveStateMachineContext()`
That component should not have hidden dependencies. I can only agree for `AaveContext` dependency

File: features/multiply/aave/components/AaveMultiplyPositionData.tsx [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+  debtTokenPrice: BigNumber
+  collateralTokenReserveData: PreparedAaveReserveData
+  debtTokenReserveData: PreparedAaveReserveData
+}
+
+const getLTVRatioColor = (ratio: BigNumber) => {
+  const critical = new BigNumber(5)
+  const warning = new BigNumber(20)
+
+  if (ratio.isLessThanOrEqualTo(critical)) {
```
Comment (jakubswierczek) [inline features/multiply/aave/components/AaveMultiplyPositionData.tsx:31]: make switch statement

File: features/multiply/aave/components/AaveMultiplyPositionData.tsx [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+  currentPosition,
+  nextPosition,
+  collateralTokenPrice,
+  debtTokenPrice,
+  collateralTokenReserveData,
+  debtTokenReserveData,
+}: AaveMultiplyPositionDataProps) {
+  const { t } = useTranslation()
+
+  const { collateral, debt, category, riskRatio } = currentPosition
```
Comment (jakubswierczek) [inline features/multiply/aave/components/AaveMultiplyPositionData.tsx:47]: Those calculations should be placed in a separated file. In the future moved to the library. And have some tests ;)

File: features/multiply/aave/components/AaveMultiplyPositionData.tsx [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+            //   newNetValue && {
+            //     variant: 'positive',
+            //     value: '????',
+            //   }
+            // }
+          />
+          <DetailsSectionContentCard
+            title={t('system.net-value')}
+            value={`${formatAmount(netValue, 'USD')} USDC`}
+            // PnL is dependant on the graph work from Damian - can be kept out of scope for now (and be hidden from UI)
```
Comment (jakubswierczek) [inline features/multiply/aave/components/AaveMultiplyPositionData.tsx:149]: remove that comment, It's unnecessary

**Review by jakubswierczek — CHANGES_REQUESTED** [review]

File: features/earn/aave/components/AavePositionHeader.tsx (hunk 1 above)
Comment (marcinciarka) [reply features/earn/aave/components/AavePositionHeader.tsx:126]: Fixed, we will ned to list the tokens available in `aaveReserveConfiguration` and `CollateralTokenTypeList` + `DebtTokenTypeList` in the future.

File: features/multiply/aave/components/AaveMultiplyManageComponent.tsx (hunk 2 above)
Comment (marcinciarka) [reply features/multiply/aave/components/AaveMultiplyManageComponent.tsx:17]: Done.

File: features/multiply/aave/components/AaveMultiplyPositionData.tsx (hunk 6 above)
Comment (marcinciarka) [reply features/multiply/aave/components/AaveMultiplyPositionData.tsx:149]: Done.

File: features/multiply/aave/components/AaveMultiplyManageComponent.tsx (hunk 3 above)
Comment (marcinciarka) [reply features/multiply/aave/components/AaveMultiplyManageComponent.tsx]: Fixed, moved the machine context out of this component and passed current one through props.

File: features/multiply/aave/components/AaveMultiplyPositionData.tsx (hunk 5 above)
Comment (marcinciarka) [reply features/multiply/aave/components/AaveMultiplyPositionData.tsx:51]: Resolving as we will move this to the library.

**Pull request merged**